### PR TITLE
Kubernetes secrets

### PR DESCRIPTION
**Pull request by fydai**
This series of pull requests move secrets configuration away from kubernetes masters and into Jenkins, as it applies the other kubernetes configurations.
https://github.com/ocf/mastodon/pull/4
https://github.com/ocf/utils/pull/109
https://github.com/ocf/metabase/pull/7

### CODE REVIEW

File: modules/ocf_jenkins/manifests/init.pp [hunk 1]
```diff
@@ -17,6 +17,14 @@
     stage => first,
   }
 
+  file {
```
Comment (abizer) [inline modules/ocf_jenkins/manifests/init.pp]: logically shouldn't this be `/etc/ocf-kubernetes/secrets` since it's reasonable to expect there may be other k8s-related things we may want on machines of this class in the future

File: modules/ocf_puppet/manifests/puppetserver.pp [hunk 2]
```diff
@@ -39,12 +39,12 @@
...
   }
 
-  # Allow kubernetes masters to access kubernetes secrets
+  # Allow Jenkins to access kubernetes secrets
   puppet_authorization::rule { 'kubernetes-secrets':
     match_request_path   => '^/puppet/v3/file_(content|metadata)s?/kubernetes-secrets$',
     match_request_type   => 'regex',
     match_request_method => ['get'],
-    allow                => suffix(lookup('kubernetes::master_nodes'), '.ocf.berkeley.edu'),
+    allow                => 'reaper.ocf.berkeley.edu',
```
Comment (abizer) [inline modules/ocf_puppet/manifests/puppetserver.pp]: use the service CNAME

File: modules/ocf_kubernetes/manifests/master.pp [hunk 3]
```diff
@@ -62,21 +62,6 @@
       mode   => '0755';
   }
 
-  # These are needed because puppetlabs-kubernetes sets the permissions to 600
-  # but the certsign script, running under kubernetes-ca, needs to access them
-
-  File['/etc/kubernetes/pki'] {
```
Comment (abizer) [inline modules/ocf_kubernetes/manifests/master.pp]: why is this being removed?

File: modules/ocf_jenkins/manifests/init.pp (hunk 1 above)
Comment (fydai) [reply modules/ocf_jenkins/manifests/init.pp]: whoops that's a typo thanks

File: modules/ocf_kubernetes/manifests/master.pp [hunk 4]
```diff
@@ -62,21 +62,6 @@
...
-  }
-
-  File['/etc/kubernetes/pki/ca.crt'] {
-    owner  => 'kubernetes-ca',
-  }
-
-  File['/etc/kubernetes/pki/ca.key'] {
-    owner  => 'kubernetes-ca',
-  }
-
```
Comment (dkess) [inline modules/ocf_kubernetes/manifests/master.pp]: Why did you remove these? Either way, it belongs in a separate commit.

File: modules/ocf_kubernetes/manifests/master.pp (hunk 3 above)
Comment (fydai) [reply modules/ocf_kubernetes/manifests/master.pp]: Somehow both me and dkess wrote a fix to the kubernetes-ca permissions problem. His solution is further down and is much better than mine.

File: modules/ocf_puppet/manifests/puppetserver.pp (hunk 2 above)
Comment (jvperrin) [inline modules/ocf_puppet/manifests/puppetserver.pp]: Does this work with `jenkins.ocf.berkeley.edu` instead, or does it have to be the fqdn of the host? It would be better if this was the service name to be more generic rather than the specific hostname of the jenkins host.

Comment (fydai) [reply modules/ocf_puppet/manifests/puppetserver.pp]: puppet doesn't like that

Comment (jvperrin) [reply modules/ocf_puppet/manifests/puppetserver.pp]: lol abizer just mentioned this too, guess I'm slow

File: modules/ocf_jenkins/manifests/init.pp [hunk 5]
```diff
@@ -17,6 +17,14 @@
...
+    '/etc/ocf-kubernetes-secrets':
+      ensure => 'directory',
+      source => 'puppet:///kubernetes-secrets',
+      purge  => true,
+      mode   => '0700';
+  }
```
Comment (dkess) [inline modules/ocf_jenkins/manifests/init.pp]: You might need to make this owned by the `jenkins-deploy` user.

File: modules/ocf_puppet/manifests/puppetserver.pp (hunk 2 above)
Comment (fydai) [reply modules/ocf_puppet/manifests/puppetserver.pp]: It needs to be the fqdn, I tested jenkins.ocf.berkeley.edu first

File: modules/ocf_kubernetes/manifests/master.pp (hunk 3 above)
Comment (dkess) [reply modules/ocf_kubernetes/manifests/master.pp]: This might still be needed, I think we added this when `puppetlabs-kubernetes` got upgraded and changed how file permissions work. Either way it should happen in a separate commit.

Comment (fydai) [reply modules/ocf_kubernetes/manifests/master.pp]: They are still needed, I just tested on deadlock.
